src/main/java/sparksoniq/jsoniq/runtime/iterator/postfix/PredicateClosureZipped.java

```diff
@@ -55,7 +55,12 @@ public Boolean call(Tuple2<Item, Long> v1) throws Exception {
         _expression.open(dynamicContext);
         Item result = _expression.next();
         _expression.close();
-        return result.getBooleanValue();
+        if(result.isNumeric())
+        {
+        	return new Boolean(result.equals(dynamicContext.getPosition()));
```

Comment:
Now follows the semantics of filters

https://www.w3.org/TR/xquery-30/#id-filter-expression